app/src/composables/use-local-storage.ts

```diff
@@ -1,38 +1,43 @@
 import { ref, watch } from 'vue';
+import { parseJSON } from '@directus/shared/utils';
 
-export function useLocalStorage(key: string) {
+type LocalStorageObjectType = string | number | boolean | object | null;
+
+export function useLocalStorage(key: string, defaultValue: LocalStorageObjectType = null) {
 	const internalKey = `directus-${key}`;
-	const data = ref<string | number | boolean | object | null>(null);
+	const data = ref<LocalStorageObjectType>(null);
+
+	function getValue(): LocalStorageObjectType {
+		const rawExistingValue = localStorage.getItem(internalKey);
+
+		if (!rawExistingValue) return defaultValue;
 
-	function getExistingValue() {
-		let rawExistingValue;
 		try {
-			rawExistingValue = localStorage.getItem(internalKey);
-		} catch (err: any) {
-			//
-		}
+			return parseJSON(rawExistingValue);
+		} catch (e) {
+			// eslint-disable-next-line no-console
+			console.warn(`Couldn't parse value from local storage`, e);
 
-		if (!rawExistingValue) return;
+			return defaultValue;
+		}
+	}
 
+	function setValue(value: LocalStorageObjectType) {
 		try {
-			const existingValue = JSON.parse(rawExistingValue);
-			data.value = existingValue;
-		} catch (err: any) {
-			//
+			localStorage.setItem(internalKey, JSON.stringify(value));
+		} catch (e) {
+			// eslint-disable-next-line no-console
+			console.warn(`Couldn't stringify and set value to local storage`, e);
 		}
 	}
 
-	getExistingValue();
+	data.value = getValue();
 
 	watch(data, () => {
-		try {
-			if (data.value == null) {
-				localStorage.removeItem(internalKey);
-			} else {
-				localStorage.setItem(internalKey, JSON.stringify(data.value));
-			}
-		} catch (err: any) {
-			//
+		if (data.value == null) {
+			localStorageObject.clear();
```

Comment:
Hey @goth-pl, thanks for this pull request!

Sorry to bother you, but just stumbled upon this while working on other stuff...
Shouldn't this be `localStorage.removeItem(internalKey)` as it used to be before? Am I missing something here?
Probably, the test needs to be updated as well.

Thanks again!

Reply:
Looks like @paescuj is right on the money here, as this line currently errors out due to `localStorageObject` not being declared anywhere.

I believe it was missed during the removal of `localStorageObject` in this commit: https://github.com/directus/directus/pull/15861/commits/d7715afabe1d3adbf8361114f770737044fecd56.

> Shouldn't this be `localStorage.removeItem(internalKey)` as it used to be before?

Seems like originally `localStorageObject` has a method called `clear()` which then actually calls `localStorage.removeItem()` so it was technically still correct: 
https://github.com/directus/directus/blob/5c8ba4b8eb7506ba6d714912834008304ad0124d/app/src/utils/local-storage-object.ts#L40-L42

Definitely confused me as well 😄 

@goth-pl Let's make sure to use `localStorage.removeItem(internalKey)` for the follow up fix 👍 (or do feel free to let us know if you don't currently have the bandwidth for this)

Reply:
Just recalled about this earlier, so I've followed this up with a PR in #16581 👍

Reply:
> Just recalled about this earlier, so I've followed this up with a PR in #16581 👍

Nice! Exactly what I was thinking of 🥇

Reply:
@paescuj @azrikahar OMG! You're right! I must have missed it while cleaning up. Sorry, thanks for catching and fixing it ❤️